linux_os/guide/system/software/integrity/software-integrity/aide/aide_periodic_cron_checking/bash/shared.sh

```diff
@@ -2,9 +2,13 @@
 
 {{{ bash_package_install("aide") }}}
 
-if ! grep -q "/usr/sbin/aide --check" /etc/crontab ; then
-    echo "05 4 * * * root /usr/sbin/aide --check" >> /etc/crontab
-else
-    sed -i '/^.*\/usr\/sbin\/aide --check.*$/d' /etc/crontab
-    echo "05 4 * * * root /usr/sbin/aide --check" >> /etc/crontab
+CRONTAB=/etc/crontab
+CRONDIRS='/etc/cron.d /etc/cron.daily /etc/cron.weekly /etc/cron.monthly'
+
+if [ -f /var/spool/cron/root ]; then
+	VARSPOOL=/var/spool/cron/root
+fi
+
+if ! grep -qR '^.*\/usr\/sbin\/aide\s*\-\-check.*|.*\/bin\/mail\s*-s\s*".*"\s*root@.*$' $CRONTAB $VARSPOOL $CRONDIRS; then
+	echo '/usr/sbin/aide --check | /bin/mail -s "$HOSTNAME - Daily aide integrity check run" root@localhost' > /etc/cron.daily/aide
```

Comment:
I think that mail notifications are already covered by the other rule (aide_scan_notification).

All artefacts within a rule such as OVAL definition, prose in rule.yml, Bash, Ansible, test scenarios and OCIL must but consistent. Over long term, we are trying to eradicate inconsistencies. This change will cause that the text in rule.yml says different thing that what the bash script does. Can you take a look at it?

Reply:
Also, at a minimum aide should run daily, at a maximum, weekly.

Reply:
The problem is that we had two different rules to ensure a check was performed, where one indicated a notification and the other didn't.

That resulted in two scan aide checks being performed daily. These aide checks are exhuastive, especially when you have McAfee VirusScan running in the background.

With the proposed change, only one daily check is performed and includes the notification, which in turn will satisfy both rules.

Reply:
Do we have an answer to this? I don't understand this area, but it looks like a legitimate concern.